main.c

```diff
@@ -836,6 +837,10 @@ int __attribute__((used)) main(void) {
     // Start the debug serial
     serial_early_init();
 
+    // Set up anything that might need to get done before we try to use SPI flash
+    // This is needed for some boards where flash relies on GPIO setup to work
+    external_flash_setup();
```

Comment:
Guard this, for boards that don't have external flash. I think this will do it -- sorry I did not test it before.. Otherwise on boards like Trinket M0, there will be link errors (as you see in the CI). Note that RP2040 is idiosyncratic because it has external flash, but it's considered to be an "internal" filesystem. See `supervisor.mk` which determines whether to compile the `external_flash/` sources or not.
```suggestion
    #if !INTERNAL_FLASH_FILESYSTEM
    external_flash_setup();
    #endif
```

Reply:
Thanks, that was it. I've made the change and tested it on a few boards. And now I know to check `supervisor.mk`!